packages/bot/src/scratch/hooks/block.js

```diff
@@ -112,3 +105,18 @@ Blockly.Block.getDimensions = function(block_node) {
 
     return block_hw;
 };
+Blockly.Block.prototype.isMainBlock = function() {
+    return config.mainBlocks.some(block_type => block_type === this.type);
+};
+
+Blockly.Block.prototype.isIndependentBlock = function() {
+    const independent_blocks = [
+        'block_holder',
+        'tick_analysis',
+        'loader',
+        'procedures_defreturn',
+        'procedures_defnoreturn',
+    ];
```

Comment:
Any reason why this is not in `const.js` file?

Reply:
I've moved these values to the constants file.